packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -15,6 +15,8 @@ import { ISource, SourceConfig } from './source';
 // eslint-disable-next-line no-duplicate-imports, import/order
 import { Construct as CoreConstruct } from '@aws-cdk/core';
 
+// tag value has a limit of 256 characters
```

Comment:
But this is part of the _key_, which is limited to 128 characters. In any case, how do you enforce this limit, given that the `destinationKeyPrefix` property doesn't seem to have any length constraint?

Reply:
Good call. I added some validation which will throw an error if the tag key is > 128 characters (`destinationKeyPrefix` can't be > 104 characters). That is a _very_ long key prefix so I don't think we will ever hit that limit, but I also can't think of a better way of handling this.